x/fileurl.go

```diff
@@ -0,0 +1,110 @@
+package x
+
+import (
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/ory/x/logrusx"
+)
+
+// winPathRegex is a regex for [DRIVE-LETTER]:
+var winPathRegex = regexp.MustCompile("^[A-Za-z]:.*")
+
+// GetURLFilePath returns the path of a URL that is compatible with the runtime os filesystem
+func GetURLFilePath(u *url.URL) string {
+	if u == nil {
+		return ""
+	}
+
+	if u.Scheme != "file" && u.Scheme != "" {
+		return u.Path
+	}
+	fPath := u.Path
+	if runtime.GOOS == "windows" {
+		if u.Host != "" {
+			// Make UNC Path
+			s := string(filepath.Separator)
+			fPath = s + s + u.Host + filepath.FromSlash(fPath)
+			return fPath
+		}
+		if winPathRegex.MatchString(fPath[1:]) {
+			// On Windows we should remove the initial path separator in case this
+			// is a normal path (for example: "\c:\" -> "c:\"")
+			fPath = stripFistPathSeparators(fPath)
+		}
+	}
+	return filepath.FromSlash(fPath)
+}
+
+// ParseURL parses rawurl into a URL structure with special handling for file:// URLs
+func ParseURL(rawurl string) (*url.URL, error) {
+	lcRawurl := strings.ToLower(rawurl)
+	if strings.Index(lcRawurl, "file:///") == 0 {
+		return url.Parse("file:///" + toSlash(rawurl[8:]))
+	}
+	if strings.Index(lcRawurl, "file://") == 0 {
+		// Normally the first part after file:// is a hostname, but since
+		// this is often misused we interpret the URL like a normal path
+		// by removing the "file://" from the beginning
+		rawurl = rawurl[7:]
+	}
+	if winPathRegex.MatchString(rawurl) {
```

Comment:
I think so because this is about parsing what is in the config file, and the config file can look the same on any system. It is only when we want to use the Path from the URL that the platform becomes a factor. However, when a POSIX system tries to use such a path it will probably return an error saying it cannot find /C:/path/file.ext. I think this is reasonable to expect.

Reply:
Yes, if this results in `/C:/path/file.ext` being used as the file path then that is what I would expect as well :+1: